### ISSUE TITLE
Terraform provider Crashing

### ISSUE THREAD

**Issue by pputman-clabs** [okta/terraform-provider-okta#1340]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
This is ran through github actions, but terraform version is latest, and I upgraded to 3.37.0 for okta terraform provider version.

### Affected Resource(s)

- okta_app_group_assignments

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```
The child module:
```terraform
resource "okta_app_group_assignments" "group_assignments" {
  for_each   = var.group_assignments
  app_id     = okta_app_saml.saml_app.id
  depends_on = [okta_app_saml.saml_app]
  lifecycle {
    ignore_changes = [group]
  }

  group {
    id = each.value.id
  }
}
```

The parent module that calls this:

```terraform
data "okta_groups" "slack_groups" {
  q = "app#slack"
}

module "okta_app_saml_slack" {
  source              = "../../terraform-modules/saml-app"
  label               = var.slack_label
  preconfigured_app   = var.slack_okta_app_name
  auto_submit_toolbar = var.slack_auto_submit_toolbar
  group_assignments   = { for group in data.okta_groups.slack_groups.groups : group.name => group }
}
```

### Debug Output

This is in github actions so I don't have the actual debug output, if its needed I can do it, but its going to take some reconfiguring to run terraform locally, so let me know if I need to do so.
https://gist.github.com/pputman-clabs/1eff99c14d621fbe562c69bc8dd2be1e

### Panic Output

https://gist.github.com/pputman-clabs/1eff99c14d621fbe562c69bc8dd2be1e

### Expected Behavior

Should have generated a plan
### Actual Behavior

API doesn't respond, and provider crashes
### Steps to Reproduce

1. `terraform apply`

### Important Factoids
It's github actions, but other plans that don't utilize this resource are working, so its not a permissions issue to okta.  Also, this plan was working last week, so not sure if anything has changed in the api?

### References
I

- #0000

**Comment by monde** [comment]
Thanks @pputman-clabs I'll get some panic bumpers on this in the next release.

Okta internal reference: https://oktainc.atlassian.net/browse/OKTA-544516

**Comment by pputman-clabs** [comment]
Thank you!  Is there any ETA I can report back to my team?

**Comment by monde** [comment]
fix released as v3.38.0 https://github.com/okta/terraform-provider-okta/releases/tag/v3.38.0

**Comment by pputman-clabs** [comment]
Tested and confirmed this is working, thank you!
